core/src/debt.ts

```diff
@@ -0,0 +1,237 @@
+    return new BigNumber(auctionsQuantityBinary._hex).shiftedBy(-MKR_NUMBER_OF_DIGITS);
+};
+
+export const getDebtAuctionBidDecreaseCoefficient = memoizee(_getDebtAuctionBidIncreaseCoefficient, {
+    promise: true,
+    length: 3,
+});
+
+const getDebtAuctionState = async (network: string, earliestEndDate: Date, greatestBid: number) => {
+    const isBidExpired = (await getNetworkDate(network)) > earliestEndDate;
+    const haveBids = greatestBid !== 0;
+    if (haveBids) {
+        if (isBidExpired) {
+            return 'ready-for-collection';
+        }
+        return 'have-bids';
+    }
+    if (isBidExpired) {
+        return 'requires-restart';
+    }
+    return 'just-started';
+};
+
+const _getContractAuctionDuration = async (network: string) => {
+    const contract = await getContract(network, CONTRACT);
+    return await contract.tau();
+};
+const getDebtContractAuctionDuration = memoizee(_getContractAuctionDuration, {
+    maxAge: 24 * 60 * 60 * 1000,
+    promise: true,
+    length: 1,
+});
+
+export const fetchDebtAuctionByIndex = async (
+    network: string,
+    auctionIndex: number
+): Promise<SurplusAuction | DebtAuction> => {
+    const contract = await getContract(network, CONTRACT);
+    const auctionData = await contract.bids(auctionIndex);
+    const isAuctionCollected = new BigNumber(auctionData.end).eq(0);
+    const auctionDuration = await getDebtContractAuctionDuration(network);
+    const fetchedAt = new Date();
+    const baseAuctionInfo: CompensationAuctionBase = {
+        network,
+        id: auctionIndex,
+        fetchedAt,
+    };
+
+    if (isAuctionCollected) {
+        const auctionLastIndex = await getDebtAuctionLastIndex(contract);
+        if (auctionLastIndex < auctionIndex) {
+            throw new Error('No active auction exists with this id');
+        }
+        return { ...baseAuctionInfo, state: 'collected' };
+    }
+
+    const auctionEndDate = new Date(auctionData.end * 1000);
+    const auctionStartDate = new Date(auctionEndDate.getTime() - auctionDuration * 1000);
+    const bidEndDate = auctionData.tic ? new Date(auctionData.tic * 1000) : undefined;
+    const earliestEndDate = bidEndDate ? getEarliestDate(auctionEndDate, bidEndDate) : auctionEndDate;
+    const state = await getDebtAuctionState(network, earliestEndDate, auctionData.tic);
+
+    return {
+        ...baseAuctionInfo,
+        earliestEndDate,
+        bidAmountMKR: new BigNumber(auctionData.bid._hex).div(WAD),
+        receiveAmountDAI: new BigNumber(auctionData.lot._hex).div(RAD),
+        receiverAddress: auctionData.guy,
+        auctionEndDate,
+        bidEndDate,
+        state,
+        fetchedAt,
+        auctionStartDate,
+    };
+};
+
+export const getActiveDebtAuctionOrUndefined = async (network: string, auctionIndex: number) => {
+    const auction = await fetchDebtAuctionByIndex(network, auctionIndex);
+    if (auction.state === 'collected') {
+        return;
+    }
+    return auction;
```

Comment:
Do you want to directly typecast the output to be to be active auction, instead of typecasting every time you're using this function?